app/v1/controllers/requestParsers/validators/BSASValidator.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers.validators
+
+import javax.inject.Inject
+
+import config.FixedConfig
+import utils.CurrentDateProvider
+import v1.controllers.requestParsers.validators.validations._
+import v1.models.domain.BSAS
+import v1.models.errors.{MtdError, RuleIncorrectOrEmptyBodyError, RuleTaxYearNotSupportedError}
+import v1.models.request.TriggerBsasRawData
+
+class BSASValidator @Inject()(val currentDateProvider: CurrentDateProvider) extends Validator[TriggerBsasRawData] with FixedConfig {
+
+  private val validationSet = List(parameterFormatValidation, bodyFormatValidator, taxYearValidator,selfEmploymentIdValidator,
+    endDateFieldValidator, startDateFieldValidator, typeOfBusinessValidator, otherBodyFieldsValidator)
+
+  private def parameterFormatValidation: TriggerBsasRawData => List[List[MtdError]] = { data =>
+    List(
+      NinoValidation.validate(data.nino)
+    )
+  }
+
+  private def bodyFormatValidator: TriggerBsasRawData => List[List[MtdError]] = { data =>
+    List(
+      JsonFormatValidation.validate[BSAS](data.body.json, RuleIncorrectOrEmptyBodyError)
+    )
+  }
+
+  private def endDateFieldValidator: TriggerBsasRawData => List[List[MtdError]] = { data =>
+    val req = data.body.json.as[BSAS]
+    List(
+      EndDateValidation.validate(req.endDate)
+    )
+  }
+
+  private def startDateFieldValidator: TriggerBsasRawData => List[List[MtdError]] = { data =>
```

Comment:
why do we need both **endDateFieldValidator** and **startDateFieldValidator**? better make it as one **DateFieldValidator**